src/pages/GetStarted/CustomWorkspaceTab/DevfileSelector.tsx

```diff
@@ -57,41 +63,44 @@ export class DevfileSelectorFormGroup extends React.PureComponent<Props, State>
     this.devfileLocationRef = React.createRef();
   }
 
-  private onDevfileSelected(meta: che.DevfileMetaData): void {
+  private onDevfileClear(): void {
+    if (this.props.onClear) {
+      this.props.onClear();
+    }
+  }
+
+  private async onDevfileSelected(meta: che.DevfileMetaData): Promise<void> {
     // clear location input
     this.devfileLocationRef.current?.clearInput();
-
     try {
-      this.loadDevfile(meta.links.self);
-    } catch {
-      // noop
+      const devfileContent = await this.props.requestDevfile(meta.links.self);
+      const devfile = safeLoad(devfileContent);
+      this.props.onDevfile(devfile);
+    } catch (e) {
+      console.warn('Failed to load devfile.', e);
+      this.showAlert({
+        key: 'load-devfile-failed',
+        title: 'Failed to load devfile.',
```

Comment:
Does it make sense to show a user a real cause of the issue? Like try to extract error message from e, like is it err cert issue, 404, 503?

Reply:
It is a general message for all cases now. It is more like an explanation why the list is empty.
We do not implement a possibility to catch err cert issues yet.

Reply:
I'll add an error message from e.

Reply:
done